pkg/storegateway/bucket.go

```diff
@@ -993,6 +1041,12 @@ func (s *BucketStore) LabelNames(ctx context.Context, req *storepb.LabelNamesReq
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
+	stats.update(func(stats *queryStats) {
+		stats.blocksQueried = len(sets)
+		stats.streamingSeriesExpandPostingsDuration += time.Since(begin)
+	})
+	s.metrics.seriesBlocksQueried.Observe(float64(len(sets)))
```

Comment:
I think this belongs better in the `record*CallResult` functions. I see it is done in the same way in the `streamingSeriesSetForBlocks` function, but I think that was a mistake. Can you move both usages to `record*CallResult`?